### PR TITLE
PostgreSQL database

### PR DESCRIPTION
**Pull request by young-rob**
## Description
This PR implements the PostgreSQL database which our application will use. The changes to the terraform files mean that upon deployment, an instance of both the app and a Postgres backing service will be created in Gov PaaS, and bound to each other.

It is also possible to run the app locally and connect to either a locally running PostgreSQL database, or one running on Gov PaaS (see the readme).

## Motivation and Context
This allows us to store Hosts, Sessions and Members in the database.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by jkempster34** [comment]
@robyoung91 Looks good to me :muscle: (bear in mind I know nothing about Terraform or anything)

Is it possible to move the migrations back to `Data/Migrations`? I think you would just need to delete the `Migrations` folder, then run: `Add-Migration InitialCreate -OutputDir "Data/Migrations"`, then the rest of the migrations should follow

**Comment by young-rob** [comment]
> @robyoung91 Looks good to me 💪 (bear in mind I know nothing about Terraform or anything)
> 
> Is it possible to move the migrations back to `Data/Migrations`? I think you would just need to delete the `Migrations` folder, then run: `Add-Migration InitialCreate -OutputDir "Data/Migrations"`, then the rest of the migrations should follow

Good idea - just had a go at doing that and have pushed the changes

**Comment by jkempster34** [comment]
> > @robyoung91 Looks good to me 💪 (bear in mind I know nothing about Terraform or anything)
> > Is it possible to move the migrations back to `Data/Migrations`? I think you would just need to delete the `Migrations` folder, then run: `Add-Migration InitialCreate -OutputDir "Data/Migrations"`, then the rest of the migrations should follow
> 
> Good idea - just had a go at doing that and have pushed the changes

Seems to have broken it 😬, I don't mind if you revert, just thought it might be nice to have all of the database-y stuff together.

Have approved either way, not sure if you want someone else who knows about this kind of stuff to review as well?

**Review by jkempster34 — APPROVED** [review]

**Comment by young-rob** [comment]
> > > @robyoung91 Looks good to me 💪 (bear in mind I know nothing about Terraform or anything)
> > > Is it possible to move the migrations back to `Data/Migrations`? I think you would just need to delete the `Migrations` folder, then run: `Add-Migration InitialCreate -OutputDir "Data/Migrations"`, then the rest of the migrations should follow
> > 
> > 
> > Good idea - just had a go at doing that and have pushed the changes
> 
> Seems to have broken it 😬, I don't mind if you revert, just thought it might be nice to have all of the database-y stuff together.
> 
> Have approved either way, not sure if you want someone else who knows about this kind of stuff to review as well?

Seems to be working alright now - I think if we redo a migration we have to delete the postgres service manually and deploy again - in this case the jobs seem to be failing until the database is created again (takes 5-10 mins). Thanks for checking it out - I'll let Steven have a quick look before merging it on Monday probably because I don't think he's in today

File: terraform/workspace-variables/dev_app_env.yml
Comment (xswoodsx) [inline terraform/workspace-variables/dev_app_env.yml]: I think any reference to rails can be deleted, this looks like leftovers from pinching the ECF stuff

File: terraform/workspace-variables/review_app_env.yml
Comment (xswoodsx) [inline terraform/workspace-variables/review_app_env.yml]: same as above on this - rails references can be removed

**Review by xswoodsx — APPROVED** [review]
Other than the comment to remove the references to rails it looks good :)

**Pull request merged**
